ably/rest_client.go

```diff
@@ -470,12 +490,29 @@ func (p *HTTPPaginatedResponse) HasNext(ctx context.Context) bool {
 	return p.nextLink != ""
 }
 
-// Items unmarshals the current page of results as JSON into the provided
+// Items unmarshals the current page of results into the provided
 // variable.
 //
 // See the "Paginated results" section in the package-level documentation.
 func (p *HTTPPaginatedResponse) Items(dst interface{}) error {
-	return json.Unmarshal(p.items, dst)
+	typ, _, err := mime.ParseMediaType(p.PaginatedResult.res.Header.Get("Content-Type"))
+	if err != nil {
+		return err
+	}
+
+	/// Turn non arrays into arrays so it fits the output kind
+	if typ == "application/json" {
+		token, _ := json.NewDecoder(bytes.NewReader(p.items)).Token()
+		if token != json.Delim('[') {
+			p.items = append(raw{'['}, p.items...)
+			p.items = append(p.items, ']')
+		}
+	} else if typ == "application/x-msgpack" {
+		if (p.items[0]&0xf0 != 0x90) && p.items[0] != 0xdc && p.items[0] != 0xdd {
```

Comment:
Needs a comment for this if statement line, something like 
"Check the first byte to see if it's one of message packs three array types."